src/types/server_urls/server_urls_bucket.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{
+    constants::{SERVER_URL_BUCKET_MAX_ITEMS, STREAMING_SERVER_URL},
+    types::profile::UID,
+};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use url::Url;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq)]
+pub struct ServerUrlsBucket {
+    /// User ID
+    pub uid: UID,
+    /// HashMap where key is `Url`, value is `DateTime<Utc>`
+    pub items: HashMap<Url, DateTime<Utc>>,
+}
+
+impl ServerUrlsBucket {
+    /// Create a new `ServerUrlsBucket` with the base URL inserted.
+    pub fn new(uid: UID) -> Self {
+        let mut items = HashMap::new();
+        let base_url: &Url = &STREAMING_SERVER_URL;
+        let mtime = Self::current_timestamp();
+        items.insert(base_url.clone(), mtime);
+
+        ServerUrlsBucket { uid, items }
+    }
+
+    fn current_timestamp() -> DateTime<Utc> {
+        chrono::Utc::now()
+    }
+
+    /// Add a new URL to the bucket.
+    pub fn add_url(&mut self, url: Url) {
+        let mtime = Self::current_timestamp();
+        self.merge_items(vec![(url, mtime)]);
```

Comment:
The use of this function seems unnecessary and could just be replaced by this since the key is the Url
```
self.items.insert(url, mtime);
```